modules/core/src/main/scala/gem/Location.scala

```diff
@@ -11,7 +11,7 @@ import scalaz.Ordering.{EQ, GT, LT}
   * `Location` is a proper prefex of another, then it sorts ahead of the other
   * `Location`.
   */
-sealed trait Location {
+sealed trait Location extends Product with Serializable {
```

Comment:
I'm open to suggestions.  It's used to sort steps.  The absolute value is meaningless but you can compare one with another to know which comes first and can always find a value that falls between any two others.  In the Logoot paper it's called a line id.